m4/cf3_platforms.m4

```diff
@@ -39,3 +39,4 @@ AM_CONDITIONAL([HPUX], [test -n "`(echo ${target_os} | egrep 'hpux|hp-ux')`"])
 AM_CONDITIONAL([FREEBSD], [test -n "`(echo ${target_os} | grep freebsd)`"])
 AM_CONDITIONAL([NETBSD], [test -n "`(echo ${target_os} | grep netbsd)`"])
 AM_CONDITIONAL([XNU], [test -n "`(echo ${target_os} | grep darwin)`"])
+AM_CONDITIONAL([SHUNIT], [test -n "`which shunit2`"])
```

Comment:
Outcome from IRC conversation: Worth attempting to get this working cross-platform if it isn't too tricky, so I will take a stab at that using Solaris and then will find someone to check it on HP UX for me.

Failing that, the tests should be limited to platforms they work on (FreeBSD, OS X, Linux?) if shunit2 is installed. Failing _that_ we should only run them on platform=FreeBSD + shunit2 available.